judge/runguard.c

```diff
@@ -828,8 +831,7 @@ void setrestrictions()
 		free(path);
 
 		if ( chroot(".")!=0 ) error(errno,"cannot change root to `%s'",cwd);
-		/* Just to make sure and satisfy Coverity scan: */
-		if ( chdir("/")!=0 ) error(errno,"cannot chdir to `/' in chroot");
+		if ( chdir(rootchdir)!=0 ) error(errno,"cannot chdir to `%s' in chroot", rootchdir);
```

Comment:
We should check here that `rootchdir` is an absolute path. Part of the reason for doing the `chdir` is to make sure that we're now inside the chroot. I'm not sure what the effect would be if we were not inside the chroot and follow a relative path...

Reply:
Or just prepend “/“ to its value in this call of chdir

Reply:
Or keep the original chdir(/) and then call the chdir(rootchdir) after it